### PR TITLE
[FIX] fix plotting of an image with color bar when maximum value is exactly zero

### LINKED ISSUE #4203: [BUG] `make_glm_report` fails in `examples/04_glm_first_level/plot_fiac_analysis.py` with `height_control="fdr"`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

For example one of the following:
- Linux Ubuntu 22.04

### Python version

- [ ] 3.12
- [X] 3.11
- [ ] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

dev version

### Expected behavior

`make_glm_report` should run with all height_control or give more meaningful errors

### Current behavior & error messages

This is what I got:

```bash
# Paste the error message here
Traceback (most recent call last):
  File "/home/remi/github/nilearn/nilearn/examples/04_glm_first_level/plot_fiac_analysis.py", line 170, in <module>
    report = make_glm_report(fmri_glm,
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/glm_reporter.py", line 230, in make_glm_report
    all_components = _make_stat_maps_contrast_clusters(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/glm_reporter.py", line 778, in _make_stat_maps_contrast_clusters
    stat_map_svg = _stat_map_to_svg(
                   ^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/glm_reporter.py", line 941, in _stat_map_to_svg
    stat_map_plot = plot_stat_map(
                    ^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/plotting/img_plotting.py", line 1289, in plot_stat_map
    display = _plot_img_with_bg(
              ^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/plotting/img_plotting.py", line 268, in _plot_img_with_bg
    display.add_overlay(
  File "/home/remi/github/nilearn/nilearn/nilearn/plotting/displays/_slicers.py", line 316, in add_overlay
    self._show_colorbar(
  File "/home/remi/github/nilearn/nilearn/nilearn/plotting/displays/_slicers.py", line 538, in _show_colorbar
    ticks = _get_cbar_ticks(cbar_vmin, cbar_vmax, offset, nb_ticks)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/plotting/displays/_slicers.py", line 823, in _get_cbar_ticks
    if offset is not None and offset / vmax > 0.12:
                              ~~~~~~~^~~~~~
ZeroDivisionError: float division by zero

```

### Steps and code to reproduce bug

```python
# run the example but use this at line 170
report = make_glm_report(fmri_glm,
                         contrasts,
                         bg_img=mean_img_,
                         height_control="fdr",
                         )
```

This was adapted from `nilearn/reporting/_visual_testing/_glm_reporter_visual_inspection_suite_.py`

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4203 

Changes proposed in this pull request:

- bump stack level of several warnings (make sure they "point" to some public API function)
- add regression test and fix

### CODE REVIEW

**Comment by bthirion** [comment]
Oops, doc does not build.
LGTM otherwise.

**Comment by Remi-Gau** [comment]
na it's my bad

I meant to open this to show the bug and then fix it: should have been a draft PR

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,8 @@
 *.swo
 *#
 
+generated_glm_reports
```
Comment (ymzayek) [inline .gitignore:10]: to keep with some kind of order, can you move this down to be with the nilearn related stuff? maybe under a `# test` section

**Review by ymzayek — COMMENTED** [review]
So far so good

**Comment by Remi-Gau** [comment]
given that the example tweaked in this PR will be nuked by #3191 I will probably not change the example file in this PR

**Comment by bthirion** [comment]
Then we need to couple the two merges because the example is broken.

**Comment by Remi-Gau** [comment]
OK the bug exists in the PR #3191 if I use height_control="fdr"

**Comment by Remi-Gau** [comment]
'bonferroni' correction is affected too

**Review by ymzayek — APPROVED** [review]
LGTM thx @Remi-Gau

**Review by bthirion — COMMENTED** [review]
LGTM. Just beware the codecov complaint.

**Review by bthirion — APPROVED** [review]

**Pull request merged**